### PR TITLE
feat(studentrefinementpage): add student refinement page

### PR DESCRIPTION
**Pull request by jinlee93**
[sc-209033]

### Summary:
- adds the student refinement page
- there are a lot of double classname styles due to nested styling of table, which needs to be cleaned up
  - [clean up ticket](https://app.shortcut.com/czi-edu/story/224979)
- clean up will block landing of this pr and until styles are less specific
### Test Plan:
- Manual testing of the page:
  - In storybook, works responsively
  - Components work and render appropriately
    - Exception: Filter drawer as filtering functionality is available in FilterDrawer story  

https://user-images.githubusercontent.com/86632227/195959809-2e653892-df1a-406d-8a9f-19016c0bfcfb.mov

### CODE REVIEW

File: .storybook/pages/StudentRefinement/StudentRefinement.stories.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { StoryObj, Meta } from '@storybook/react';
+
+import { StudentRefinement } from './StudentRefinement';
+
+export default {
+  title: 'Pages/SDT/StudentRefinement',
+  component: StudentRefinement,
+} as Meta<Args>;
```
Comment (dierat) [inline .storybook/pages/StudentRefinement/StudentRefinement.stories.ts:8]: Can we get a mobile view of this page?

File: .storybook/pages/StudentRefinement/StudentRefinement.module.css [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  .student-refinement__name-cell.student-refinement__name-cell {
+    display: grid;
+    grid-template-columns: 7rem auto;
+    align-items: center;
+    margin-bottom: 0;
+  }
+
+  .student-refinement__name-cell-text.student-refinement__name-cell-text {
+    @mixin eds-theme-typography-body-text-sm;
+    font-weight: 400;
```
Comment (dierat) [inline .storybook/pages/StudentRefinement/StudentRefinement.module.css:137]: What's the font weight before adding this line?

File: src/components/TableCell/TableCell.tsx [hunk 3]
```diff
@@ -81,7 +80,6 @@ export const TableCell = ({
   const componentClassName = clsx(
     styles['table__cell'],
     align === 'left' && styles['table__cell--align-left'],
-    align === 'insetLeft' && styles['table__cell--align-inset-left'],
```
Comment (dierat) [inline src/components/TableCell/TableCell.tsx:84]: Should `insetLeft` be removed from the type definition on line 10?

**Review by dierat — APPROVED** [review]
LGTM just a couple small questions

File: .storybook/pages/StudentRefinement/StudentRefinement.stories.ts (hunk 1 above)
Comment (jinlee93) [reply .storybook/pages/StudentRefinement/StudentRefinement.stories.ts:8]: https://user-images.githubusercontent.com/86632227/199840873-5b7c87f8-2978-4be4-a1f4-d1921a658154.mov

File: .storybook/pages/StudentRefinement/StudentRefinement.module.css (hunk 2 above)
Comment (jinlee93) [reply .storybook/pages/StudentRefinement/StudentRefinement.module.css:137]: 400, will remove line

File: .storybook/pages/StudentRefinement/StudentRefinement.stories.ts (hunk 1 above)
Comment (dierat) [reply .storybook/pages/StudentRefinement/StudentRefinement.stories.ts:8]: Sorry, I meant can we get a chromatic test covering the mobile view of this page.

File: .storybook/pages/StudentRefinement/StudentRefinement.tsx [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+import { DataSummaryCard } from '../../recipes/DataSummaryCard/DataSummaryCard';
+import { PageShell } from '../../recipes/PageShell/PageShell';
+
+export const StudentRefinement = () => (
+  <PageShell>
+    <Breadcrumbs>
+      <Breadcrumbs.Item href="#" text="My Courses" />
+    </Breadcrumbs>
+    <div className={styles['student-refinement__header']}>
+      <PageHeader title="Page title" />
```
[the remark this answers is not in the document]
Comment (jinlee93) [reply .storybook/pages/StudentRefinement/StudentRefinement.tsx:35]: Yup will do

File: src/components/TableCell/TableCell.tsx (hunk 3 above)
Comment (jinlee93) [reply src/components/TableCell/TableCell.tsx:84]: yes, removed

File: .storybook/pages/StudentRefinement/StudentRefinement.stories.ts (hunk 1 above)
Comment (jinlee93) [reply .storybook/pages/StudentRefinement/StudentRefinement.stories.ts:8]: Oh ya, definitely. Added some chromatic breakpoints

**Review by dierat — APPROVED** [review]

**Pull request merged**